src/Driver.cpp

```diff
@@ -562,17 +604,59 @@ static void FindNodesByName(
     bool found = false;
 
     // Try all named nodes first
+    bool foundMatchingPrefix = false;
+    bool prefixIsAmbigious = false;
+    const NamedNodeData* nodeDataForMatchingPrefix = nullptr;
+    struct StringWithScore
+    {
+      StringWithScore(int _score, const char* _string) : score(_score), string(_string) {}
+      int score;
+      const char* string;
+    };
+    std::vector<StringWithScore> fuzzyMatches;
```

Comment:
have you tested for perf hit on a large dag (e.g. tiny)?

Reply:
I did not, only tried it on baselib
However, this only whole apparatus only applies to named nodes so unless tiny creates thousands of aliases this should not be a problem, right? Baselib on windows has 653 named nodes/aliases and it seemed okay, but did not measure the impact before vs after. I assumed so far that this won't be noticable unless it goes into the thousands but then the Dag is likely huge as well in which case I _assume_ this means that the PathCompares we do when we don't find a named node should still be much slower.

Reply:
... gonna do a project with 10000 random generated aliases and check what's the practical perf difference, now I'm curious :)

Can you confirm my general assumption that only aliases are relevant for this?

Reply:
Numbers!

Created a test application with 100000 aliases, each a random string of length 12, then run bee with different tundra versions, each time requesting a unidentifiable target (= worst case)

Full runtime on my windows machine (build graph already generated):
Before: 205 - 210ms
After: 268 - 296ms

I think this is quite acceptable (and much better than I thought) :)

Reply:
Beautiful!

Reply:
1000 aliases:
Before: 298-205ms
After: 195-208ms

btw. measured in powershell by running `Measure-Command { ./bee blubblub }` a couple of times